lib/mongo/protocol.ex

```diff
@@ -27,12 +27,18 @@ defmodule Mongo.Protocol do
   defp connect(opts, s) do
     # TODO: with/else in elixir 1.3
     result =
-      with {:ok, s} <- tcp_connect(opts, s),
-           {:ok, s} <- wire_version(s),
-           {:ok, s} <- Mongo.Auth.run(opts, s) do
+      with {:ok, s} <- tcp_connect(opts, s) do
+        inner_result = if opts[:skip_auth] do
+          {:ok, s}
```

Comment:
We don't need wire_version if we skip auth? When is skip_auth used, can we change it to checking if username and password is set?